Neighborly/VectorList.cs

```diff
@@ -149,39 +127,8 @@ public List<Vector> GetAllVectors()
 
     public void Insert(int index, Vector item)
     {
-        if (index < 0 || index > Count)
-        {
-            throw new ArgumentOutOfRangeException(nameof(index), index, "Index is out of range");
-        }
-
-        lock (_lock)
-        {
-            // Check if this item can be inserted in memory
-            if (index < _maxInMemoryCount)
-            {
-                // There is room in the in-memory list
-                _vectorList.Insert(index, item);
-                _onDiskFilePaths.Insert(index, string.Empty); // Empty string indicates that the item is in memory
-                _guids.Insert(index, item.Id);
-            }
-            else
-            {
-                // There is no room in the in-memory list
-                // Store the item on disk
-
-                var path = Path.GetTempFileName();
-
-                _vectorList.Insert(index, null);    // null indicates that the item is on disk
-                _onDiskFilePaths.Insert(index, path);
-                _guids.Insert(index, item.Id);
-
-                SaveToDisk(item, path);
-
-            }
-
-        }
-        Modified?.Invoke(this, EventArgs.Empty);
-
+        // TODO: Implement Insert
+        throw new NotImplementedException();
```

Comment:
How important is the `Insert` method as use-case for the vector DB? Generally speaking, index-based (especially `int`-based) methods don't seem to be too important for the API.

Reply:
We can omit index-based insertion and deletion altogether. I initially kept it while I was experimenting with storage. 

The order is unimportant, except in the ultra-rare use cases: time series analysis data, which, IMO, isn't a suitable candidate for a Vector-only database.

Reply:
That's what I had thought, and why I was a bit confused about the index-based methods existing at all. I was wondering if there might be any use case for search algorithms where it might be useful, but then again, those usually provide their own scoring.